cmd/snap-confine/mount-support.c

```diff
@@ -333,11 +334,34 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 						 config->rootfs_dir, dir);
 				sc_must_snprintf(dst, sizeof dst, "%s%s",
 						 scratch_dir, dir);
-				sc_do_mount(src, dst, NULL, MS_BIND, NULL);
-				sc_do_mount("none", dst, NULL, MS_SLAVE, NULL);
+				if (access(src, F_OK) == 0
+				    && access(dst, F_OK) == 0) {
+					sc_do_mount(src, dst, NULL, MS_BIND,
+						    NULL);
+					sc_do_mount("none", dst, NULL, MS_SLAVE,
+						    NULL);
+				}
 			}
 		}
 	}
+	if (config->uses_base_snap) {
+		// snapd dir
+		const char *src = "/usr/lib/snapd";
```

Comment:
This will clash with libexecdir layout.

Reply:
Indeed, I changed the code now to use the libexecdir configure option. However I think this needs to be even smarter, i.e. we need to take re-exec into account as well. I added a fixme. we could of course mount `dirname(sys.argv[0])` which is probably the easiest and most sensible thing to do.